src/Query/Transformer/NamedArrayParameter.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Linio\Component\Database\Query\Transformer;
+
+use Linio\Component\Database\Exception\InvalidQueryException;
+use Linio\Component\Database\Query\Transformer;
+
+class NamedArrayParameter implements Transformer
+{
+    public function execute(string &$query, array &$params = []): void
+    {
+        foreach ($params as $paramKey => $paramValue) {
+            $isNumeric = is_numeric($paramKey);
+            $isArray = is_array($paramValue);
+
+            if ($isNumeric && $isArray) {
+                throw new InvalidQueryException(sprintf('Unnamed parameter "%s" can\'t have array value.', $paramKey));
+            }
+
+            if ($isNumeric || !$isArray) {
```

Comment:
Move this condition to be the first one, since it would be the most common.

Reply:
actually, I noticed it could be even more simple:

```php
        foreach ($params as $paramKey => $paramValue) {
            if (!is_array($paramValue)) {
                continue;
            }

            if (is_numeric($paramKey)) {
                // .... throw exception ...
            }

            // ... transform $query and $params here ...
        }
```